src/NLog/Targets/EventLogTarget.cs

```diff
@@ -131,8 +130,26 @@ public EventLogTarget(IAppDomain appDomain)
         [DefaultValue("Application")]
         public string Log { get; set; }
 
+        private int m_MaxMessageLength;
+        /// <summary>
+        /// Gets or sets the message length limit to write to the Event Log.
+        /// </summary>
+        /// <remarks><value>MaxMessageLength</value> cannot be zero or negative</remarks>
+        [DefaultValue(16384)]
+        public int MaxMessageLength
+        {
+            get { return this.m_MaxMessageLength; }
+            set
+            {
+                if (value <= 0)
+                    throw new ArgumentException("MaxMessageLength cannot be zero or negative.");
```

Comment:
I think that there are two different set posibilities. One is the configuration initialization, other is directly set.

Therefore I choose this way. By this way;
1. NLogConfigurationException will throw at configuration initialization.
2. ArgumentException will throw at directly setting. Because it is not about configuration, I guess.

(Unfortunately I could not find any same thing at other targets.)